### ISSUE TITLE
Why create_colocated_worker_cls and spawn 

### ISSUE THREAD

**Issue by eelxpeng** [volcengine/verl#29]
It seems that there are two ways to make two worker group to use the same resource pool:
1. Separate Workers on Same Resource Pool:
```python
# Creates two separate worker groups sharing resources
actor_wg = RayWorkerGroup(resource_pool=resource_pool, ray_cls_with_init=actor_cls)
critic_wg = RayWorkerGroup(resource_pool=resource_pool, ray_cls_with_init=critic_cls)
```
2. Colocated Workers:
```python
# Creates a single worker group that implements both actor and critic
cls_dict = {'actor': actor_cls, 'critic': critic_cls}
ray_cls_with_init = create_colocated_worker_cls(cls_dict)
wg_dict = RayWorkerGroup(resource_pool=resource_pool, ray_cls_with_init=ray_cls_with_init)

# Spawns interfaces to access different functionalities
spawn_wg = wg_dict.spawn(prefix_set=cls_dict.keys())
```

With the latest main branch of Ray package where the fix https://github.com/ray-project/ray/pull/48088 is merged, the first method should work without any problem to allow two different worker group to re-use the same GPUs. It appears that the 2nd method is not that straightforward. But the PPO implementation uses the second method to put all worker groups to the global pool. Is there any specific reason for preferring the 2nd method? Thanks.

**Comment by PeterSH6** [comment]
Hi @eelxpeng , thanks for your question.

Yes, you are right that the latest branch fixed this bundle issue and we are able to implement the first case with `RayResourcePool(..., max_colocate=5)` to support Separate Workers on the Same Resource Pool. We'll check the functionality of the latest Ray release and update veRL accordingly.

To discuss the two methods. 
The first case is particularly useful for the Megatron-LM backend. Different model may require distinct 3D parallel groups, so each model needs to be stored in a separate WorkerGroup.
On the other hand, the second case is highly efficient for the FSDP backend deployed with Colocate Placement. Colocating different models within the same set of processes can save unnecessary storage of multiple CUDA contexts for different models in different WorkerGroup Processes. This approach can save GPU memory usage.

In our initial release, the bundle issue in Ray was not fixed. As a result, we implemented PPO using the second method with `RayResourcePool(..., max_colocate=1)`. We will make these two options selectable later once we've checked the functionality of the latest Ray release.
